block_darwin.go

```diff
@@ -0,0 +1,250 @@
+	VendorName   string `plist:"Vendor Name"`
+}
+
+func (ctx *context) getDiskUtilListPlist() (diskUtilListPlist, error) {
+	out, err := exec.Command("diskutil", "list", "-plist").Output()
+	if err != nil {
+		return diskUtilListPlist{}, errors.Wrap(err, "diskutil list failed")
+	}
+
+	var data diskUtilListPlist
+	if _, err := plist.Unmarshal(out, &data); err != nil {
+		return diskUtilListPlist{}, errors.Wrap(err, "diskutil list plist unmarshal failed")
+	}
+
+	return data, nil
+}
+
+func (ctx *context) getDiskUtilInfoPlist(device string) (diskUtilInfoPlist, error) {
+	out, err := exec.Command("diskutil", "info", "-plist", device).Output()
+	if err != nil {
+		return diskUtilInfoPlist{}, errors.Wrapf(err, "diskutil info for %q failed", device)
+	}
+
+	var data diskUtilInfoPlist
+	if _, err := plist.Unmarshal(out, &data); err != nil {
+		return diskUtilInfoPlist{}, errors.Wrapf(err, "diskutil info plist unmarshal for %q failed", device)
+	}
+
+	return data, nil
+}
+
+func (ctx *context) getIoregPlist(ioDeviceTreePath string) (ioregPlist, error) {
+	elems := strings.Split(ioDeviceTreePath, "/")
+	name := elems[len(elems)-1]
+
+	args := []string{
+		"ioreg",
+		"-a",      // use XML output
+		"-d", "1", // limit device tree output depth to root node
+		"-r",       // root device tree at matched node
+		"-n", name, // match by name
+	}
+	out, err := exec.Command(args[0], args[1:]...).Output()
+	if err != nil {
+		return ioregPlist{}, errors.Wrapf(err, "ioreg query for %q failed", ioDeviceTreePath)
+	}
+
+	var data []ioregPlist
+	if _, err := plist.Unmarshal(out, &data); err != nil {
+		return ioregPlist{}, errors.Wrapf(err, "ioreg unmarshal for %q failed", ioDeviceTreePath)
+	}
+	if len(data) != 1 {
+		err := errors.Errorf("ioreg unmarshal resulted in %d I/O device tree nodes (expected 1)", len(data))
+		return ioregPlist{}, err
+	}
+
+	return data[0], nil
+}
+
+func (ctx *context) makePartition(disk, s diskOrPartitionPlistNode, isAPFS bool) (*Partition, error) {
+	if s.Size < 0 {
+		return nil, errors.Errorf("invalid size %q of partition %q", s.Size, s.DeviceIdentifier)
+	}
+
+	var partType string
+	if isAPFS {
+		partType = "APFS Volume"
+	} else {
+		partType = s.Content
+	}
+
+	info, err := ctx.getDiskUtilInfoPlist(s.DeviceIdentifier)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Partition{
+		Disk:       nil, // filled in later
+		Name:       s.DeviceIdentifier,
+		Label:      s.VolumeName,
+		MountPoint: s.MountPoint,
+		SizeBytes:  uint64(s.Size),
+		Type:       partType,
+		IsReadOnly: !info.WritableVolume,
+	}, nil
+}
+
+func (ctx *context) blockFillInfo(info *BlockInfo) error {
+	listPlist, err := ctx.getDiskUtilListPlist()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err.Error())
+		return err
+	}
+
+	info.TotalPhysicalBytes = 0
+	info.Disks = make([]*Disk, 0, len(listPlist.AllDisksAndPartitions))
+	info.Partitions = []*Partition{}
+
+	for _, disk := range listPlist.AllDisksAndPartitions {
+		if disk.Size < 0 {
```

Comment:
Ditto. If `Size` was a `uint64` this wouldn't be necessary.

Reply:
See above.